Neos.Flow/Classes/Composer/InstallerScripts.php

```diff
@@ -121,10 +121,11 @@ protected static function copyDistributionFiles(string $installerResourcesDirect
      * Calls a static method from it's string representation
      *
      * @param string $staticMethodReference
+     * @param PackageEvent $event
      * @return void
      * @throws Exception\InvalidConfigurationException
      */
-    protected static function runPackageScripts(string $staticMethodReference): void
+    protected static function runPackageScripts(string $staticMethodReference, PackageEvent $event): void
```

Comment:
because this method is protected, this is a potentially breaking change.. Not very likely that someone overrode this, but maybe it should be marked as such to make sure?

Reply:
Id say this is def not public api because `InstallerScripts` is not api, this is not `public` and frankly i see no way to extend this class and use it?

And more importantly we use `self::runPackageScripts` vs `static::runPackageScripts` which makes it completely impossible to override this part ;)

Reply:
I agree that it's not likely. But the `InstallerScripts` class is no final and this method is `protected` – so code that did
```php
class MyInstallerScripts extend InstallerScripts {
  protected static function runPackageScripts(string $staticMethodReference): void {
    InstallerScripts::runPackageScripts($staticMethodReference);
    // my custom logic
  }
}
```
would now break.

Another reason to use `final` classes and `private` methods by default.

Anyways, not blocking this.
Not +1ing it either, because I don't understand the usecase fully yet

Reply:
The usecase to pass the event is following: I've a `neos-build` package who copy files to root folder, but it ask the user if a) it should copy it, b) override existing files and c) if the user use JavaScript or TypeScript. As `neos-build` packages can't have flow commands, it is done via installer scripts. In the future there will be some more options like which type of CSS framework they want to use and if the root should be the project or a custom package (If you want to develop site package shipped will all settings to build the frontend assets)

Without the event, it is not possible to check, if the install script is running in non-interactive mode